### PR TITLE
feature(pipeline): queue polish the logic to set needReRangePendingQueue flag

### PR DESCRIPTION
**Pull request by sfwn**
#### What type of this PR

feature

#### What this PR does / why we need it:

Queue set needReRangePendingQueue flag if newItem's order is higher than the current ranging one.

### CODE REVIEW

File: modules/pipeline/pipengine/queue/priorityqueue/priority_queue.go [hunk 1]
```diff
@@ -94,3 +94,21 @@ func (pq *PriorityQueue) Range(f func(Item) (stopRange bool)) {
...
+		return true
+	}
+	if !leftExist {
+		return false
+	}
+	if !rightExist {
+		return true
+	}
+	// both exist, judge by index
+	return leftItem.Index() < rightItem.Index()
```
Comment (kakj-go) [inline modules/pipeline/pipengine/queue/priorityqueue/priority_queue.go:113]: only leftItem.Index() < rightItem.Index() return true other return false

**Review by kakj-go — APPROVED** [review]

File: modules/pipeline/pipengine/queue/priorityqueue/priority_queue.go (hunk 1 above)
Comment (sfwn) [reply modules/pipeline/pipengine/queue/priorityqueue/priority_queue.go:113]: if right item doesn't exists, return true.

**Pull request merged**